app/models/post.rb

```diff
@@ -1,5 +1,21 @@
 class Post < ApplicationRecord
-  has_many :likes, foreign_key: :post_id
-  has_many :comments, foreign_key: :post_id
-  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  after_save :update_posts_counter
+
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_counter, comparison: { greater_than_or_equal_to: 0 }, numericality: { only_integer: true }
+  validates :likes_counter, comparison: { greater_than_or_equal_to: 0 }, numericality: { only_integer: true }
+
+  def last_five_comments
+    comments.order('created_at Desc').limit(5)
+  end
+
+  private
+
+  def update_posts_counter
+    author.update(postCounter: author.posts.length)
```

Comment:
- **[OPTIONAL]** I think we can make this shorter by using the `increment!` method.

  ```
  def update_posts_counter
    author.increment!(:postCounter)
  end
  ```

Reply:
thanks, @DammyShittu  for your great suggestions I will apply them right now